pkg/installmanager/installmanager.go

```diff
@@ -1260,9 +1260,17 @@ func updateClusterProvisionWithRetries(provision *hivev1.ClusterProvision, m *In
 }
 
 func cleanupLogOutput(fullLog string) string {
-	var cleanedString string
-
-	cleanedString = multiLineRedactLinesWithPassword.ReplaceAllString(fullLog, "REDACTED LINE OF OUTPUT")
+	var cleanedString = fullLog
+
+	// The console log may have carriage returns as well as newlines,
+	// and they may be escaped (this especially happens with the
+	// baremetal IPI installer when libvirt.so emits errors). Unescape
+	// the returns and newlines, then remove the carriage returns, so
+	// that the string is treated as multiple lines.
+	cleanedString = strings.ReplaceAll(cleanedString, "\\r", "\r")
+	cleanedString = strings.ReplaceAll(cleanedString, "\\n", "\n")
```

Comment:
This should be done with `strconv.Unquote` instead.

Reply:
Unquote requires manipulating the input string to wrap it in quotes first, and the code here ends up being less obvious. I'll change it if you want, but I think this is clearer.

Reply:
Here are my concerns with your current approach.
1) It only unescapes a small selection of things. If we are going to unescape then we should unescape everything.
2) It is not correct in all cases. For example, see https://play.golang.org/.
3) It is adding custom code to do something for which there is already well-tested code.

Reply:
I think the URL for that sample case is missing some parts? I just get the default playground.

I made the change to Unquote(), and it broke other tests. I've modified the error messages in those tests to show the text that was found so it can be compared to what was expected.

```
--- FAIL: Test_builder_Build (120.03s)
    --- FAIL: Test_builder_Build/no_override (30.01s)
        remoteclient_test.go:218:
            	Error Trace:	remoteclient_test.go:218
            	Error:      	"Get https://api.hive-cluster.example.com:6443/api?timeout=32s: dial tcp: i/o timeout" does not contain "no such host"
            	Test:       	Test_builder_Build/no_override
            	Messages:   	expected to find "no such host" in error, got "Get https://api.hive-cluster.example.com:6443/api?timeout=32s: dial tcp: i/o timeout"
    --- FAIL: Test_builder_Build/override_active (30.01s)
        remoteclient_test.go:218:
            	Error Trace:	remoteclient_test.go:218
            	Error:      	"Get https://url-override/api?timeout=32s: dial tcp: i/o timeout" does not contain "no such host"
            	Test:       	Test_builder_Build/override_active
            	Messages:   	expected to find "no such host" in error, got "Get https://url-override/api?timeout=32s: dial tcp: i/o timeout"
    --- FAIL: Test_builder_Build/no_override,_dynamic (30.01s)
        remoteclient_test.go:218:
            	Error Trace:	remoteclient_test.go:218
            	Error:      	"Get https://api.hive-cluster.example.com:6443/apis/hive.openshift.io/v1/ClusterDeployment/bad-name: dial tcp: i/o timeout" does not contain "no such host"
            	Test:       	Test_builder_Build/no_override,_dynamic
            	Messages:   	expected to find "no such host" in error, got "Get https://api.hive-cluster.example.com:6443/apis/hive.openshift.io/v1/ClusterDeployment/bad-name: dial tcp: i/o timeout"
    --- FAIL: Test_builder_Build/override_active,_dynamic (30.01s)
        remoteclient_test.go:218:
            	Error Trace:	remoteclient_test.go:218
            	Error:      	"Get https://url-override/apis/hive.openshift.io/v1/ClusterDeployment/bad-name: dial tcp: i/o timeout" does not contain "no such host"
            	Test:       	Test_builder_Build/override_active,_dynamic
            	Messages:   	expected to find "no such host" in error, got "Get https://url-override/apis/hive.openshift.io/v1/ClusterDeployment/bad-name: dial tcp: i/o timeout"
FAIL
```

I get that we want this to be flexible and always do the right thing. I think the easiest way to do that is to just fix the parts of the strings that are actually causing problems with the regex currently, and iterate if we find other issues.

Reply:
That test failure is unrelated to the changes that you are making. It is failing because of your DNS settings. The test is fragile and relies on DNS not resolving the domain url-override.

Reply:
I've pushed a new version using Unquote().